app/src/main/java/org/mozilla/fenix/share/ShareViewModel.kt

```diff
@@ -87,15 +104,44 @@ class ShareViewModel(application: Application) : AndroidViewModel(application) {
         }
     }
 
+    private fun filterRecentApps(
+        apps: List<AppShareOption>,
+        recentApps: List<AppShareOption>
+    ): List<AppShareOption> {
+        val leastRecentApps = ArrayList<AppShareOption>()
+        for (app in apps) {
+            if (!recentApps.contains(app)) {
+                leastRecentApps.add(app)
+            }
+        }
+        return leastRecentApps
+    }
+
+    @WorkerThread
+    internal fun buildRecentAppsList(apps: List<AppShareOption>): List<AppShareOption> {
+        val recentAppsResult = ArrayList<AppShareOption>()
+        val recentAppsDatabase = recentAppsStorage.getRecentAppsUpTo(RECENT_APPS_LIMIT)
+
+        for (recentApp in recentAppsDatabase) {
```

Comment:
Instead of the double for loop maybe a filter could work here? Something like 
`apps.filter { a -> recentAppsDatabase.any { it.name == a.name } }`? Not 100% sure if that's the same so up to you!

Reply:
Filter is actually perfect for this case, I replaced the loops. Thank you!

Reply:
In the end, had to revert back to the double for loop. The way filter works is by preserving the order of apps, but actually I wanted to use the order of recentApps, since that one is correctly sorted. 
I tried another solution with filter and and a map/find pair, but it was even harder to understand, so I kept this one